main/webservices/api/test_api.php

```diff
@@ -2,116 +2,531 @@
+        return $data;
+    }
 
-/**
- * @param $apiKey
- *
- * @throws Exception
- *
- * @return int
- */
-function createUser($apiKey)
-{
-    global $webserviceURL;
-    global $webserviceUsername;
-    $client = new Client([
-        'base_uri' => $webserviceURL,
-    ]);
-
-    $response = $client->post(
-        'v2.php',
-        [
-            'form_params' => [
-                // data for the user who makes the request
-                'action' => 'save_user',
-                'username' => $webserviceUsername,
-                'api_key' => $apiKey,
-                // data for new user
-                'firstname' => 'Test User',
-                'lastname' => 'Chamilo',
-                'status' => 5, // student
-                'email' => 'testuser@example.com',
-                'loginname' => 'restuser',
-                'password' => 'restuser',
-                'original_user_id_name' => 'myplatform_user_id', // field to identify the user in the external system
-                'original_user_id_value' => '1234', // ID for the user in the external system
-                'extra' => [
-                    [
-                        'field_name' => 'age',
-                        'field_value' => 29,
-                    ],
+    /**
+     * Posts an action request and assert the server returns an array with a single element
+     *
+     * @param string    $action         name of the webservice action
+     * @param array     $parameters     parameters to send with the request
+     * @return mixed                    the unique element of the "data" array
+     */
+    protected function singleElementValue($action, $parameters = [])
+    {
+        $data = $this->dataArray($action, $parameters);
+        $this->assertSame(1, count($data));
+        return $data[0];
+    }
+
+    /**
+     * Posts an action request and assert it returns an array with a single integer value
+     *
+     * @param string    $action         name of the webservice action
+     * @param array     $parameters     parameters to send with the request
+     * @return integer                  the integer value
+     */
+    protected function integer($action, $parameters = [])
+    {
+        $value = $this->singleElementValue($action, $parameters);
+        $this->assertIsInt($value);
+        return $value;
+    }
+
+    /**
+     * Posts an action request and assert it returns an array with a single boolean value
+     *
+     * @param string    $action         name of the webservice action
+     * @param array     $parameters     parameters to send with the request
+     * @return boolean                  the boolean value
+     */
+    protected function boolean($action, $parameters = [])
+    {
+        $value = $this->singleElementValue($action, $parameters);
+        $this->assertIsBool($value);
+        return $value;
+    }
+
+    /**
+     * @throws Exception if it cannot delete the created test user
+     */
+    public function testCreateUser()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $EXTRA_FIELD_VALUE is not named in camelCase.](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)